lager/lenses/tuple.hpp

```diff
@@ -0,0 +1,194 @@
+#pragma once
+
+#include <tuple>
+#include <type_traits>
+
+#include <lager/lenses.hpp>
+#include <lager/lenses/attr.hpp>
+
+namespace lager {
+namespace lenses {
+namespace detail {
+
+template <typename T>
+struct templated_type_contructor;
+
+template <template <typename...> typename T, typename... Params>
+struct templated_type_contructor<T<Params...>>
+{
+    template <typename... Ts>
+    static auto construct(Ts&&... ps)
+    {
+        return T<std::remove_reference_t<Ts>...>{std::forward<Ts>(ps)...};
+    }
+};
```

Comment:
It seems like the only thing this is really doing is the `remove_reference_t`. This could be done way above the call stack in a less confusing way and get rid of `templated_type_constructor`. Also not sure why you want to `remove_reference` at all...

Reply:
`templated_type_constructor` is here to match against a specific `tuple` or `pair` (could be extended to arrays but I couldn't find a good use for it), and gives me a way to construct a value of that particular tuple templated type with any arguments.

Reply:
I tried rewriting this with a member templated type instead, but it doesn't really make the code any more readable or even shorter... it's just in a kind of awkward spot.